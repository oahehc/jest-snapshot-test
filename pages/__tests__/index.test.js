/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import App from '../index.js'

describe('With Snapshot Testing', () => {
  it('App shows "Hello world!"', () => {
    const app = shallow(<App />)

    expect(app).toMatchSnapshot()
  })
})
