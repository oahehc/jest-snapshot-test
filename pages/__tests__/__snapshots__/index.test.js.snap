// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Hello world!" 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <JSXStyle
          css={
            String {
              "0": "p",
              "1": ".",
              "10": "3",
              "11": "5",
              "12": "5",
              "13": "0",
              "14": "1",
              "15": "3",
              "16": "{",
              "17": "c",
              "18": "o",
              "19": "l",
              "2": "j",
              "20": "o",
              "21": "r",
              "22": ":",
              "23": "r",
              "24": "e",
              "25": "d",
              "26": ";",
              "27": "}",
              "3": "s",
              "4": "x",
              "5": "-",
              "6": "2",
              "7": "5",
              "8": "8",
              "9": "7",
              "__hash": "2587355013",
            }
          }
          styleId="2587355013"
        />,
        <p
          className="jsx-2587355013"
        >
          Hello World!
        </p>,
      ],
      "className": "jsx-2587355013",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": String {
            "0": "p",
            "1": ".",
            "10": "3",
            "11": "5",
            "12": "5",
            "13": "0",
            "14": "1",
            "15": "3",
            "16": "{",
            "17": "c",
            "18": "o",
            "19": "l",
            "2": "j",
            "20": "o",
            "21": "r",
            "22": ":",
            "23": "r",
            "24": "e",
            "25": "d",
            "26": ";",
            "27": "}",
            "3": "s",
            "4": "x",
            "5": "-",
            "6": "2",
            "7": "5",
            "8": "8",
            "9": "7",
            "__hash": "2587355013",
          },
          "styleId": "2587355013",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello World!",
          "className": "jsx-2587355013",
        },
        "ref": null,
        "rendered": "Hello World!",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <JSXStyle
            css={
              String {
                "0": "p",
                "1": ".",
                "10": "3",
                "11": "5",
                "12": "5",
                "13": "0",
                "14": "1",
                "15": "3",
                "16": "{",
                "17": "c",
                "18": "o",
                "19": "l",
                "2": "j",
                "20": "o",
                "21": "r",
                "22": ":",
                "23": "r",
                "24": "e",
                "25": "d",
                "26": ";",
                "27": "}",
                "3": "s",
                "4": "x",
                "5": "-",
                "6": "2",
                "7": "5",
                "8": "8",
                "9": "7",
                "__hash": "2587355013",
              }
            }
            styleId="2587355013"
          />,
          <p
            className="jsx-2587355013"
          >
            Hello World!
          </p>,
        ],
        "className": "jsx-2587355013",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": String {
              "0": "p",
              "1": ".",
              "10": "3",
              "11": "5",
              "12": "5",
              "13": "0",
              "14": "1",
              "15": "3",
              "16": "{",
              "17": "c",
              "18": "o",
              "19": "l",
              "2": "j",
              "20": "o",
              "21": "r",
              "22": ":",
              "23": "r",
              "24": "e",
              "25": "d",
              "26": ";",
              "27": "}",
              "3": "s",
              "4": "x",
              "5": "-",
              "6": "2",
              "7": "5",
              "8": "8",
              "9": "7",
              "__hash": "2587355013",
            },
            "styleId": "2587355013",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hello World!",
            "className": "jsx-2587355013",
          },
          "ref": null,
          "rendered": "Hello World!",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
